web/src/pages/apd/new/ApdNew.test.js

```diff
@@ -0,0 +1,272 @@
+import React from 'react';
+import {
+  renderWithConnection,
+  act,
+  screen,
+  waitFor
+} from 'apd-testing-library';
+import { mockFlags, resetLDMocks } from 'jest-launchdarkly-mock';
+import userEvent from '@testing-library/user-event';
+
+import ApdNew from './ApdNew';
+
+jest.mock('../../../util/api', () => ({
+  get: jest.fn(),
+  post: jest.fn(),
+  patch: jest.fn(),
+  delete: jest.fn()
+}));
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: jest.fn()
+}));
+
+const defaultProps = {
+  createApd: jest.fn(),
+  years: ['2023', '2024'],
+  yearOptions: ['2023', '2024', '2025']
+};
+
+let options;
+let props;
+const setup = async (props = {}, options = {}) => {
+  let util;
+  // eslint-disable-next-line testing-library/no-unnecessary-act
+  await act(async () => {
+    util = renderWithConnection(
+      <ApdNew {...defaultProps} {...props} />,
+      options
+    );
+  });
+  const user = userEvent.setup();
+  return {
+    util,
+    user
+  };
+};
+
+describe('<ApdNew />', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.setTimeout(30000);
+    resetLDMocks();
+  });
+
+  describe('form with Mmis enabled', () => {
+    beforeEach(() => {
+      mockFlags({ enableMmis: true });
+    });
+
+    it('should display form with MMIS', async () => {
+      await setup(props, options);
+      expect(
+        screen.getByText(/Create a New Advanced Planning Document/)
+      ).toBeTruthy();
+      expect(screen.getByText(/MMIS IAPD/)).toBeTruthy();
```

Comment:
does it default to MMIS? Is it supposed to do that? I thought it defaulted to neither one selected

Reply:
I started with the test with the flag on:

```
describe('form with Mmis enabled', () => {
    beforeEach(() => {
      mockFlags({ enableMmis: true });
    });
```

No reason why that's first.